realtime/server.js

```diff
@@ -70,25 +77,26 @@ socket.on('connection', function (client) {
             sendLoginErrorResponse(username, client, 'chars');
             return;
         }
-        if (usernames[username]) {
+        if (usernames[username]) { 
             sendLoginErrorResponse(username, client, 'taken');
             return;
         }
-        people[client.id] = username;
-        usernames[username] = true;
+        people[client.id] = username; //people has value 'username' at position 'client.id'
```

Comment:
Assume the programmer knows the Javascript language. This comment is unnecessary.